### DISCUSSION TITLE
Copilot creating bugs cause of outdated training set

### DISCUSSION THREAD

**Discussion by 4skl** [community/community#82217] [Copilot Conversations]
I had problems using copilot cause the training database is old;
For example I need to use winit 0.25.0 to work with copilot else I get errors, as I use copilot to learn new skills this is a bit annoying...
What do you think of that ?

**Comment by Akash1134** [comment]
Copilot's suggestions are generated based on a vast amount of code from public repositories on which it is trained on, but it's important to note that the training data used by Copilot may not always reflect the latest updates or versions of libraries and frameworks. This can result in suggestions that may not align with the specific versions you are using. Languages with less representation in public repositories may produce fewer or less robust suggestions.

As the tool and technology used around it is growing, the suggestions will get better. 

Thanks!

**Reply by 4skl** [reply]
Thanks, in my case it was about rust and wgpu+winit, finally helped me but isn't it trained on documentations too ? (I guess only the ones on GitHub right ?)

**Reply by kurutah** [reply]
Am I correct in understanding that it updates in real time and hasn't just been frozen in the state of 2021?
